src/assets/src/hooks/useWebSocket.ts

```diff
@@ -67,6 +67,6 @@ export const useWebSocket = <T>(url: string, onUpdate: (content: T) => void, onD
             ws.onmessage = null;
             ws.close();
         }
-    }, []);
+    }, [url]);
     return error;
```

Comment:
I see this is working. 

I see this function is called from 2 places, `useQueueWebSocket` and `useUserWebSocket`, this bug might have originated when updating Queue and not user?

Would it be possible to explain what is these 2 function purpose is?

Reply:
I didn't update either of the hooks you mentioned in my frontend update (https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/405/files). These hooks are thin wrappers around the hook I'm modifying here, `useWebSocket`, which is like the websocket equivalent of `usePromise`, it manages the state around the lifecycle of a websocket connection. I believe all the wrappers do is form a URL using an ID and pass the needed callback functions.

I think I'm on the right track for explaining the cause of this problem in the issue under Commentary. I don't quite understand why migrating `react-router-dom` caused this, but perhaps it's because of the change from using `props` to a hook (`useParams`). I haven't yet tracked down a GitHub issue or anything like that to explain this.

I think adding `url` here just makes sense. It's possible it could have side effects on other parts of the application, but in general, if the ID (either for a user, or a queue) changes, it makes sense to re-initiate the Websocket connection. By adding `url`, we're subscribing to changes to that URL (and by extension the queue ID or user ID), whereas before we were not).

I looked at all the `useUserWebSocket` usages, and I don't expect the user ID passed to them to change (since it would require a logout, we're only really using this websocket to get the current user's data). And looking at all the `useQueueWebSocket` usages (which on the Settings page, the Manage Queue page, and the Queue visitor page), the only time I think the ID would change without a full re-render is in the case related to this bug here. It's possible if we added more functionality that you'd see this problem with navigating between `/manage/{queueID}` pages or `/manage/{queueUD}/settings` pages, but the fix I'm providing here would resolve those cases too.

Hope that helps.